source/module_base/test/container_operator_test.cpp

```diff
@@ -0,0 +1,120 @@
+#include "../container_operator.h"
+#include "gtest/gtest.h"
+#include <iostream>
+
+/************************************************
+ *  unit test of container operator
+ ***********************************************/
+
+/**
+ * - Tested Functions:
+ *   - VectorPlus
+ *     - "+" operator for vectors
+ *   - VectorMinus
+ *     - "-" operator for vectors
+ *   - VectorMultiply
+ *     - "*" operator for scalar*vector
+ *   - MapPlus
+ *     - "+" operator for maps
+ *   - MapMinus
+ *     - "-" operator for maps
+ *   - MapMultiply
+ *     - "*" operator for scalar*map
+ */
+
+TEST(ContainerOperator,VectorPlus)
+{
+	std::vector<double> a(20,10.0);
+	std::vector<double> b(20,1.0);
+	std::vector<double> c(20);
+	c = a+b;
+	for (int i=0;i<c.size();i++)
+		EXPECT_EQ(c[i],11.0);
+}
+
+TEST(ContainerOperator,VectorMinus)
+{
+	std::vector<double> a(20,10.0);
+	std::vector<double> b(20,1.0);
+	std::vector<double> c(20);
+	c = a-b;
+	for (int i=0;i<c.size();i++)
+		EXPECT_EQ(c[i],9.0);
+}
+
+TEST(ContainerOperator,VectorMultiply)
+{
+	std::vector<double> a(20,10.0);
+	double b = 2.0;
+	std::vector<double> c(20);
+	c = b*a;
+	for (int i=0;i<c.size();i++)
+		EXPECT_EQ(c[i],20.0);
+}
+
+TEST(ContainerOperator,LengthCheck)
```

Comment:
The name `MapLengthCheck` is better.
Is there any length check on vector operations?